usaspending_api/recipient/delta_models/sam_recipient.py

```diff
@@ -1,29 +1,90 @@
     )
     USING DELTA
     LOCATION 's3a://{{SPARK_S3_BUCKET}}/{{DELTA_LAKE_S3_PATH}}/{{DESTINATION_DATABASE}}/{{DESTINATION_TABLE}}'
     """
+sam_recipient_load_sql_string = [
+    rf"""
+CREATE OR REPLACE TEMPORARY VIEW Source_data AS (
+      SELECT
+        ROW_NUMBER() OVER ( PARTITION BY broker_sam_recipient.award_or_recipient_uniqu
+                            ORDER BY
+                            awardee_or_recipient_uniqu ASC NULLS LAST,
+                            activation_date DESC NULLS LAST
+                            ) AS row_num,
+        broker_sam_recipient.awardee_or_recipient_uniqu,
+        broker_sam_recipient.legal_business_name,
+        broker_sam_recipient.dba_name,
+        broker_sam_recipient.ultimate_parent_unique_ide,
+        broker_sam_recipient.ultimate_parent_legal_enti,
+        broker_sam_recipient.address_line_1,
+        broker_sam_recipient.address_line_2,
+        broker_sam_recipient.city,
+        broker_sam_recipient.state,
+        broker_sam_recipient.zip,
+        broker_sam_recipient.zip4,
+        broker_sam_recipient.country_code,
+        broker_sam_recipient.congressional_district,
+        COALESCE(broker_sam_recipient.business_types_codes, text[]) AS business_types_codes,
+        broker_sam_recipient.entity_structure,
+        broker_sam_recipient.sam_recipient_id,
+        COALESCE(broker_sam_recipient.activation_date, broker_sam_recipient.deactivation_date) as record_date,
+        broker_sam_recipient.uei,
+        broker_sam_recipient.ultimate_parent_uei
+      FROM
+        global_temp.sam_recipient as broker_sam_recipient
+);""",
+    rf"""INSERT OVERWRITE {{DESTINATION_DATABASE}}.{{DESTINATION_TABLE}} AS (
+  SELECT
+    broker_sam_recipient.awardee_or_recipient_uniqu AS awardee_or_recipient_uniqu,
+    broker_sam_recipient.legal_business_name AS legal_business_name,
+    broker_sam_recipient.ultimate_parent_unique_ide AS ultimate_parent_unique_ide,
+    broker_sam_recipient.ultimate_parent_legal_enti AS ultimate_parent_legal_enti,
+    broker_sam_recipient.sam_recipient_id AS broker_duns_id,
+    broker_sam_recipient.record_date AS update_date,
+    broker_sam_recipient.address_line_1 AS address_line_1,
+    broker_sam_recipient.address_line_2 AS address_line_2,
+    broker_sam_recipient.city AS city,
+    broker_sam_recipient.congressional_district AS congressional_district,
+    broker_sam_recipient.country_code AS country_code,
+    broker_sam_recipient.state AS state,
+    broker_sam_recipient.zip AS zip,
+    broker_sam_recipient.zip4 AS zip4,
+    broker_sam_recipient.business_types_codes AS business_types_codes,
+    broker_sam_recipient.dba_name as dba_name,
+    broker_sam_recipient.entity_structure as entity_structure,
+    broker_sam_recipient.uei AS uei,
+    broker_sam_recipient.ultimate_parent_uei AS ultimate_parent_uei
+  FROM
+    Source_data
+    Where Source_data.row_num !=0
```

Comment:
`If you are going to uppercase other SQL terms then do the same here with `Where`.  Also adjust the indentation to match the formatting that you have.

Reply:
`ROW_NUMBER` starts at the value 1 and so this condition will match all data. This should instead look only for those rows where `row_num = 1` since that will be the first item from the "distinct" group; matching the current Postgres `DISTINCT ON` implementation.

Reply:
Thank you for clarifying the DISTINCT ON implementation.